### DISCUSSION TITLE
Get window PID with xprop

### DISCUSSION THREAD

**Discussion by haarp** [netblue30/firejail#5967] [Q&A]
Hello,

`xprop _NET_WM_PID` allows to get the PID of the window clicked upon under X.

However this doesn't work for firejail windows, as I'm only getting very low PIDs, presumably those within the jail. They do not correspond to any PIDs seen in `firejail <--list|--tree|--top>`.

Is there any way to get a mapping from the in-jail PIDs to system-wide PIDs?

(Background: I want to SIGSTOP processes by clicking on them, powered by xprop)

Thanks!

**Comment by rusty-snake** [comment]
> Is there any way to get a mapping from the in-jail PIDs to system-wide PIDs?

If you know the jail (precisely: the pid-namespace), there could be a way. But without you try to find the parent of `2` in the following tree:

```
2463: 1, 2, 3
83644: 1, 4
1293: 2, 5
```

Short: Even if you have a mapping, the input key is not unique.

**Reply by haarp** [reply]
Hmmmm, yeah, that makes things difficult. Not seeing an easy way around this. Thanks for the info!
